code/game/machinery/hologram.dm

```diff
@@ -339,7 +342,9 @@ Possible to do for anyone motivated enough:
 				HC.Disconnect(src)//can't answer calls while calling
 			else
 				playsound(src, 'sound/machines/twobeep.ogg', 100)	//bring, bring!
+				ringing = TRUE
 
+	SetLightsAndPower()

```

Comment:
Is this going to try to update light source every process ?

Reply:
Currently, yes. I could just call update_icon().

Are repeated calls of set_light(X) expensive? I thought it wouldn't recalculate if it was the same value.

Reply:
Not horribly since it will pull out on required update check most likely, but it's still few needless procalls

Reply:
You could check to see if ringing is the same as when the proc started before doing the update.

Reply:
PLease to do the needful